assembly/src/errors.rs

```diff
@@ -255,7 +255,7 @@ impl ParsingError {
 
     pub fn malformed_doc_comment(step: usize) -> Self {
         ParsingError {
-            message: "doc comments separated by line break".to_string(),
+            message: "comment should always be followed by a procedure token".to_string(),
```

Comment:
I would change the text here to: "Procedure comment is not immediately followed by a procedure declaration." And I would change the function name to `dangling_procedure_comment()`.